### PR TITLE
Add notification context

### PR DESCRIPTION
**Pull request by sbryzak**
Fixes https://issues.redhat.com/browse/CRT-682

### CODE REVIEW

File: pkg/controller/notification/notification_context_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		Spec: v1alpha1.UserSignupSpec{
+			Username:      "jsmith@redhat.com",
+			Approved:      true,
+			TargetCluster: "east",
+			FamilyName:    "Smith",
+			GivenName:     "John",
+		},
+	}
+
+	r, _, _ := prepareReconcile(t, userSignup.Name, userSignup, basicNSTemplateTier)
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/controller/notification/notification_context_test.go:68]: I didn't realise this was unnecessary, removed.

File: pkg/controller/notification/notification_context_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+	// then
+	require.NoError(t, err)
+
+	require.Equal(t, userSignup.Name, notificationCtx.UserID)
+	require.Equal(t, userSignup.Spec.GivenName, notificationCtx.FirstName)
+	require.Equal(t, userSignup.Spec.FamilyName, notificationCtx.LastName)
+	require.Equal(t, userSignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey], notificationCtx.Email)
+	require.Equal(t, userSignup.Spec.Company, notificationCtx.CompanyName)
+}
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/controller/notification/notification_context_test.go:81]: Added

File: pkg/controller/notification/notification_context_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		Data: map[string][]byte{
+			"token": []byte("mycooltoken"),
+		},
+	}
+
+	initObjs = append(initObjs, secret)
+
+	client := test.NewFakeClient(t, initObjs...)
+
+	r := &ReconcileUserSignup{
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/controller/notification/notification_context_test.go:133]: This is an artifact from copy/pasting, I've removed the unnecessary bits.

File: pkg/controller/notification/notification_context_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+		},
+		Spec: toolchainv1alpha1.NSTemplateTierSpec{
+			Namespaces: namespaces,
+			ClusterResources: &toolchainv1alpha1.NSTemplateTierClusterResources{
+				TemplateRef: nstemplatetiers.NewTierTemplateName(tierName, "clusterresources", clusterRevision),
+			},
+		},
+	}
+}
+
```
Comment (alexeykazakov) [inline pkg/controller/notification/notification_context_test.go:49]: You don't need this function either, do you?
```suggestion

```

File: pkg/controller/notification/notification_context_test.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	client := prepareReconcile(t, userSignup.Name, userSignup)
+
+	// when
+	_, err := NewNotificationContext(context.Background(), client, "other", operatorNamespace)
+
+	// then
+	require.Error(t, err)
+}
```
Comment (alexeykazakov) [inline pkg/controller/notification/notification_context_test.go:98]: Or simply:
```suggestion
func TestNotificationContext(t *testing.T) {
	// given
	userSignup := &v1alpha1.UserSignup{
		ObjectMeta: newObjectMeta("john", "jsmith@redhat.com"),
		Spec: v1alpha1.UserSignupSpec{
			Username:      "jsmith@redhat.com",
			Approved:      true,
			TargetCluster: "east",
			FamilyName:    "Smith",
			GivenName:     "John",
		},
	}
	client := prepareReconcile(t, userSignup.Name, userSignup)

	t.Run("user found", func(t *testing.T) {
		// when
		notificationCtx, err := NewNotificationContext(context.Background(), client, userSignup.Name, operatorNamespace)

		// then
		require.NoError(t, err)

		require.Equal(t, userSignup.Name, notificationCtx.UserID)
		require.Equal(t, userSignup.Spec.GivenName, notificationCtx.FirstName)
		require.Equal(t, userSignup.Spec.FamilyName, notificationCtx.LastName)
		require.Equal(t, userSignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey], notificationCtx.Email)
		require.Equal(t, userSignup.Spec.Company, notificationCtx.CompanyName)
	})

	t.Run("user not found", func(t *testing.T) {
		// when
		_, err := NewNotificationContext(context.Background(), client, "other", operatorNamespace)

		// then
		require.Error(t, err)
	})
}
```

File: pkg/controller/notification/notification_context_test.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+		},
+	}
+
+	client := prepareReconcile(t, userSignup.Name, userSignup)
+
+	// when
+	_, err := NewNotificationContext(context.Background(), client, "other", operatorNamespace)
+
+	// then
+	require.Error(t, err)
```
Comment (alexeykazakov) [inline pkg/controller/notification/notification_context_test.go:97]: Let's check the error message too.

File: pkg/controller/notification/notification_context_test.go (hunk 4 above)
Comment (sbryzak) [reply pkg/controller/notification/notification_context_test.go:49]: Right, it's unused now. Removed.

File: pkg/controller/notification/notification_context_test.go (hunk 6 above)
Comment (sbryzak) [reply pkg/controller/notification/notification_context_test.go]: Done

**Pull request merged**